soc/xtensa/intel_adsp/tools/cavstool_client.py

```diff
@@ -133,9 +136,24 @@ def main():
 if args.quiet:
     log.setLevel(logging.WARN)
 
-HOST = args.server_addr
+if args.server_addr:
+    try:
+        HOST, PORT = args.server_addr.split(":", 2)
```

Comment:
This is not compatible with IPv6. Use `urlparse`:
```python
>>> from urllib.parse import urlparse

>>> urlparse('//[::1]:123').hostname
'::1'
>>> urlparse('//[::1]:123').port
123
>>> print(urlparse('//[::1]').port)
None
```

Reply:
This is really good for me, thanks for letting me know this!